solidity/nft_token.sol

```diff
@@ -0,0 +1,254 @@
+import "./library/spl_token.sol";
+import "./library/system_instruction.sol";
+
+struct Position {
+    address whirlpool;      // 32
+    address position_mint;  // 32
+    uint128 liquidity;      // 16
+    int32 tick_lower_index; // 4
+    int32 tick_upper_index; // 4
+}
+
+@program_id("DPvSokZKEAYqh6jjmGDUyd4L881uAWg4rFVeA6rec2fr")
+contract nft_token {
+    address public orca;
+    address public whirlpool;
+    address public pdaProgram;
+    address public mintErc20;
+    address public pdaERC20Account;
+    // TODO: Change for the latter one in production
+    uint64 public pdaHeader = 0x55bdfe33;//0xd0f7407ae48fbcaa;
+    // PDA seed
+    bytes public constant pdaProgramSeed = "pdaProgram";
+    // PDA bump
+    bytes1 public pdaBump;
+    int32 public constant min_tick_lower_index = -443632;
+    int32 public constant max_tick_lower_index = 443632;
+    //address constant tokenProgram = address"TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA";
+
+    uint64 public numTokenAccounts;
+
+    mapping(address => uint64) public mapTokenAccountLiquidity;
+    address[type(uint32).max] public tokenAccounts;
+
+    @space(10000)
```

Comment:
?

Reply:
if you are creating a PDA account, you're limited to 10kb for the account size.
space is "solang helper" for allocated space for "PDA" data account.
"Alternatively, the data account can be created by the constructor, on chain. When this method is used, some parameters must be specified for the account using annotations. Annotations placed above a constructor can only contain literals or constant expressions, as is the case for first @seed and @space"
This is a feature of Solana and such a bridge is made by Solang developers to use a language similar to Solidity along with the concept of Solana accounts.